graql/reasoner/tree/NodeImpl.java

```diff
@@ -84,9 +91,26 @@ public void addChild(Node child){
 
     @Override
     public void addAnswer(ConceptMap answer){
+        if (answer.isEmpty()) return;
+        validateAnswer(answer);
         answers.add(answer);
     }
 
+    private void validateAnswer(ConceptMap answer){
```

Comment:
why do we need this? can we write a comment for the method about why / what its doing?